CseEightselectBasic/Services/Export/Export.php

```
@@ -0,0 +1,105 @@
+<?php
+namespace CseEightselectBasic\Services\Export;
+
+use Shopware;
+
+class Export {
+  public function __construct() {
+    $container = Shopware()->Container();
+    $this->configValidator = $container->get('cse_eightselect_basic.config.validator');
+    $this->pluginConfig = $container->get('cse_eightselect_basic.dependencies.provider')->getPluginConfig();
+  }
+
+  private function getFormatParam($request) {
+    return $request->getParam('format');
+  }
+
+  private function isTidAndFidInRequest ($request) {
+    $tenantId = $request->getHeader('8select-com-tid');
+    $feedId = $request->getHeader('8select-com-fid');
+    
+    return $tenantId != null && $feedId != null;
+  }
+
+  private function isValidFormatSpecified ($request) {
+    $format = self::getFormatParam($request);
+    
+    if ($format == null) {
+      return false;
+    }
+    
+    $validFormats = ['product_feed', 'property_feed', 'status_feed'];
+    $isFormatValid = in_array($format, $validFormats, true);
+    
+    return $isFormatValid;
+  }
+
+  private function isTidAndFidValid ($request) {
+    $tenantIdMatches = $request->getHeader('8select-com-tid') === $this->pluginConfig['CseEightselectBasicApiId'];
+    $feedIdMatches = $request->getHeader('8select-com-fid') === $this->pluginConfig['CseEightselectBasicFeedId'];
+    return $tenantIdMatches && $feedIdMatches;
+  }
+
+  public function getOffsetAndLimit ($request) {
+    //@todo maybe use symfony
+    $MAX_LIMIT = 100;
+    $limit = $request->getParam('limit', 50);
+    $offset = $request->getParam('offset', 0);
+
+    if(!is_numeric($limit) || !is_numeric($offset)) {
+      return false;
+    }
+
+    $limit = intval($limit);
+    $offset = intval($offset);
+  
+    if($limit > $MAX_LIMIT) {
+      return false;
```

do we want to return an error message that the provided limit is bigger than the max. allowed limit?